torch/utils/data/datapipes/iter/routeddecoder.py

```diff
@@ -1,38 +1,47 @@
-from torch.utils.data import IterDataPipe
+import re
+
+from io import BufferedIOBase
+from typing import Any, Callable, Iterable, Iterator, Sized, Tuple
+
+from torch.utils.data import IterDataPipe, functional_datapipe
 from torch.utils.data.datapipes.utils.decoder import (
     Decoder,
     basichandlers as decoder_basichandlers,
     imagehandler as decoder_imagehandler)
 
-from typing import Iterable, Iterator, Union, List, Tuple, Any, Callable
-from io import BufferedIOBase
 
+# Extract extension from pathname
+def _default_key_fn(pathname):
+    return re.sub(r".*[.]", "", pathname)
+
+
+@functional_datapipe('decode')
 class RoutedDecoderIterDataPipe(IterDataPipe[Tuple[str, Any]]):
     r""" :class:`RoutedDecoderIterDataPipe`.
 
-    Iterable datapipe to decode binary streams from input iterables,
-    yield pathname and decoded binary stream in a tuple.
+    Iterable datapipe to decode binary streams from input DataPipe, yield pathname
+    and decoded data in a tuple.
     args:
         datapipe: Iterable datapipe that provides pathname and binary stream in tuples
-        handlers: user defined decoder handlers, if None, basic and image decoder handlers will be set as default
-        length: a nominal length of the datapipe
```

Comment:
Will the removal of the `length` parameter happen to all datapipes?

Reply:
`length` is an optional argument for all datapipes. Only for the datapipe that can not be inferred for the length based on arguments, `length` argument is needed 
For this DataPipe, the length should remain same as the input datapipe. So, no need to have length as an argument, just like all other funcitonal datapipe such as Map.